LuckyDragon/app/src/main/java/com/example/luckydragon/Activities/ViewEventActivity.java

```diff
@@ -191,17 +194,46 @@ public void loadChildFragment() {
     public void sampleAttendeesIfNeccessary() {
         if(!event.isLoaded()) return;
         if(event == null || event.getCreatedTimeMillis() == null) return;
+
+        GlobalApp globalApp = (GlobalApp) getApplication();
         if(!event.haveInviteesBeenSelected()) {
             LocalDateTime currentDateTime = LocalDateTime.now();
             LocalDate lotteryDate = LocalDate.parse(event.getLotteryDate());
             LocalTime lotteryTime = LocalTime.of(event.getLotteryHours(), event.getLotteryMinutes());
             LocalDateTime lotteryDateTime = LocalDateTime.of(lotteryDate, lotteryTime);
             if(currentDateTime.isAfter(lotteryDateTime)) {
                 event.selectInviteesFirstTime();
+
+                String notSelectedTitle = "Update re - " + event.getName();
+                String notSelectedBody = "You have unfortunately not been selected to attend " +
+                        "the '" + event.getName() + "' event. You will remain on the waitlist " +
+                        "and may possibly be selected if another entrant declines to attend.";
+                for (String deviceId : event.getWaitList()) {
+                    User notSelectedUser = globalApp.getUserById(deviceId);
+                    notSelectedUser.addToNotificationList(notSelectedTitle, notSelectedBody);
+                }
+
+                String selectedTitle = "Congratulations!";
+                String selectedBody = "You have been selected to attend the '" + event.getName() +
+                        "' event! Please open the event in the app and accept the invitation " +
+                        "for the event if you would like to attend.";
+                for (String deviceId : event.getInviteeList()) {
+                    User selectedUser = globalApp.getUserById(deviceId);
+                    selectedUser.addToNotificationList(selectedTitle, selectedBody);
+                }
             }
         } else {
             // check to see if we can fill replacement invitees if there are spots left
-            event.fillInvitees();
+            ArrayList<String> replacementInvitees = event.fillInvitees();
+
+            String replacedTitle = "Congratulations!";
+            String replacedBody = "You have been selected to attend the '" + event.getName() +
+                    "' event as another entrant has declined their invitation. Please open the " +
+                    "event in the app and accept the invitation if you would like to attend.";
+            for (String deviceId : replacementInvitees) {
+                User replacedUser = globalApp.getUserById(deviceId);
+                replacedUser.addToNotificationList(replacedTitle, replacedBody);
+            }
```

Comment:
Again, why not just make it part of the fillInvitees() function?

Reply:
This cannot be part of fillInvitees since we cannot access globalApp from within Event to get the User, I tried for many hours to do this but it does not work.